mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -740,6 +742,59 @@ def _get_metric_value_details(self, path, metric):
     def log_metric(self, run_id, metric):
         # simply call _log_metrics and let it handle the rest
         self._log_metrics(run_id, [metric], isSingleMetric=True)
+        self._log_model_metrics(run_id, [metric])
+
+    def _log_model_metrics(
+        self, run_id: str, metrics: list[Metric], path: str = "", is_single_metric: bool = False
+    ) -> None:
+        if not metrics:
+            return
+
+        metric_instances: list[SqlLoggedModelMetric] = []
+        for index, metric in enumerate(metrics):
+            if metric.model_id is None:
+                continue
+            path = path if is_single_metric else append_to_json_path(path, f"[{index}]")
+            metric, value, _is_nan = self._get_metric_value_details(path, metric)
+            metric_instances.append(
+                SqlLoggedModelMetric(
+                    model_id=metric.model_id,
+                    metric_name=metric.key,
+                    metric_timestamp_ms=metric.timestamp,
+                    metric_step=metric.step,
+                    metric_value=value,
+                    experiment_id=metric.experiment_id,
+                    run_id=run_id,
+                    dataset_uuid=metric.dataset_uuid,
+                    dataset_name=metric.dataset_name,
+                    dataset_digest=metric.dataset_digest,
+                )
+            )
+
+        with self.ManagedSessionMaker() as session:
+            try:
+                session.add_all(metric_instances)
+                session.commit()
+            except sqlalchemy.exc.IntegrityError:
+                session.rollback()
+                metric_keys = [m.metric_name for m in metric_instances]
+                metric_key_batches = (
+                    metric_keys[i : i + 100] for i in range(0, len(metric_keys), 100)
+                )
+                for batch in metric_key_batches:
+                    existing_metrics = (
+                        session.query(SqlLoggedModelMetric)
+                        .filter(
+                            SqlLoggedModelMetric.run_id == run_id,
+                            SqlLoggedModelMetric.metric_name.in_(batch),
+                        )
+                        .all()
+                    )
+                    existing_metrics = {m.to_mlflow_entity() for m in existing_metrics}
+                    non_existing_metrics = [
+                        m for m in metric_instances if m.to_mlflow_entity() not in existing_metrics
+                    ]
+                    session.add_all(non_existing_metrics)
```

Comment:
TODO: Add some comments on why we're doing this.

Reply:
Why *are* we doing this? :D

Reply:
Good question. To ignore duplicate metrics and not throw.

Reply:
Awesome - let's add an inline comment